### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by ukurere**

### CODE REVIEW

File: ChemistryLaboratory/ChemistryLaboratory/Class1.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+        public ChemistyLaboratory(int max)
+        {
+            maxSubstances = max;
+            currentSubstancesCount = 0;
+        }
+
+        public void Add()
+        {
+            if (currentSubstancesCount < maxSubstances)
```
Comment (Puchkof) [inline ChemistryLaboratory/ChemistryLaboratory/Class1.cs]: What if condition is false?

File: ChemistryLaboratory/ChemistryLaboratory/Class1.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            {
+                currentSubstancesCount++;
+            }
+        }
+
+        public void Div()
+        {
+            if (currentSubstancesCount > 0)
```
Comment (Puchkof) [inline ChemistryLaboratory/ChemistryLaboratory/Class1.cs]: What if condition is false?

File: ChemistryLaboratory/ChemistryLaboratory/Program.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
+using MyApp.Class;
+
+Console.WriteLine("Please enter the maximum number of substances on the shelf:");
+string input = Console.ReadLine();
+
+int maxSubstances;
+if (!int.TryParse(input, out maxSubstances))
+{
+    Console.WriteLine("Invalid number. Please restart the program and enter a valid number.");
```
Comment (Puchkof) [inline ChemistryLaboratory/ChemistryLaboratory/Program.cs]: I would not stop program here. Use while loop

File: ChemistryLaboratory/ChemistryLaboratory/Program.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ChemistyLaboratory lab = new ChemistyLaboratory(maxSubstances);
+
+while (true)
+{
+    Console.WriteLine("Enter command (add/div):");
+    string command = Console.ReadLine();
+
+    if (command == "add")
+    {
+        if (lab.currentSubstancesCount == maxSubstances)
```
Comment (Puchkof) [inline ChemistryLaboratory/ChemistryLaboratory/Program.cs]: I would incapsulate this logic inside lab

File: ChemistryLaboratory/ChemistryLaboratory/Program.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            Console.WriteLine("Number of substances on the shelf is max.");
+        lab.Add();
+    }
+    else if (command == "div")
+    {
+        if (lab.currentSubstancesCount == 0)
```
Comment (Puchkof) [inline ChemistryLaboratory/ChemistryLaboratory/Program.cs]: I would incapsulate this logic inside lab

File: ChemistryLaboratory/ChemistryLaboratory/Program.cs (hunk 3 above)
Comment (ukurere) [reply ChemistryLaboratory/ChemistryLaboratory/Program.cs]: But we read maxSubstances once. That's why I check it before the loop

File: ChemistryLaboratory/ChemistryLaboratory/Program.cs (hunk 4 above)
Comment (ukurere) [reply ChemistryLaboratory/ChemistryLaboratory/Program.cs]: This I am doing now

File: ChemistryLaboratory/ChemistryLaboratory/Class1.cs (hunk 1 above)
Comment (ukurere) [reply ChemistryLaboratory/ChemistryLaboratory/Class1.cs]: Nothing(?)
I don`t know what I have to write here

File: ChemistryLaboratory/ChemistryLaboratory/Class1.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+{
+    public class ChemistyLaboratory
+    {
+        public int maxSubstances;
+        public int currentSubstancesCount;
+        public string command;
+
+        public ChemistyLaboratory(int max)
+        {
+            maxSubstances = max;
```
Comment (Puchkof) [inline ChemistryLaboratory/ChemistryLaboratory/Class1.cs]: You can initialise both fields without constructor

File: ChemistryLaboratory/ChemistryLaboratory/Class1.cs (hunk 1 above)
Comment (Puchkof) [reply ChemistryLaboratory/ChemistryLaboratory/Class1.cs]: You can either return `bool` as the result of the method or throw exception, you should always notify user about something going wrong

File: ChemistryLaboratory/ChemistryLaboratory/Class1.cs (hunk 2 above)
Comment (Puchkof) [reply ChemistryLaboratory/ChemistryLaboratory/Class1.cs]: You can either return `bool` as the result of the method or throw exception, you should always notify user about something going wrong

File: ChemistryLaboratory/ChemistryLaboratory/Program.cs (hunk 3 above)
Comment (Puchkof) [reply ChemistryLaboratory/ChemistryLaboratory/Program.cs]: But what is the reason of stopping program? You need to achieve critical state to cancel program execution, not typical "invalid input" case. Just keep asking user about valid input here

File: ChemistryLaboratory/ChemistryLaboratory/Class1.cs [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+            currentSubstancesCount = 0;
+        }
+
+        public void Add()
+        {
+            if(currentSubstancesCount == maxSubstances)
+            {
+                throw new Exception("Reach the maximum limit. You can not add substances on the Stage. \n");
```
Comment (Puchkof) [inline ChemistryLaboratory/ChemistryLaboratory/Class1.cs]: I would suggest you to create custom exceptions

File: ChemistryLaboratory/ChemistryLaboratory/Class2.cs [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+namespace MyApp.Class
+{
+    public class MaxLimitExceededException : Exception
+    {
+        public MaxLimitExceededException() : base("Reach the maximum limit. You can not add substances on the Stage.")
+        {
+
+        }
+    }
+    public class NoSubstancesException : Exception
```
Comment (Puchkof) [inline ChemistryLaboratory/ChemistryLaboratory/Class2.cs]: Put different classes to different files

File: ChemistryLaboratory/ChemistryLaboratory/Class1.cs [hunk 9]
```diff
@@ -0,0 +1,42 @@
+using System;
+
+
+namespace MyApp.Class
+{
+    public class ChemistyLaboratory
+    {
+        public int maxSubstances;
+        public int currentSubstancesCount;
+        public string command;
```
Comment (Puchkof) [inline ChemistryLaboratory/ChemistryLaboratory/Class1.cs]: You dont need to store this field here, it should be just variable in Program.cs

File: ChemistryLaboratory/ChemistryLaboratory/Class1.cs [hunk 10]
```diff
@@ -0,0 +1,42 @@
+using System;
+
+
+namespace MyApp.Class
+{
+    public class ChemistyLaboratory
+    {
+        public int maxSubstances;
```
Comment (Puchkof) [inline ChemistryLaboratory/ChemistryLaboratory/Class1.cs]: Make this field as private

File: ChemistryLaboratory/ChemistryLaboratory/Class1.cs [hunk 11]
```diff
@@ -0,0 +1,42 @@
+using System;
+
+
+namespace MyApp.Class
+{
+    public class ChemistyLaboratory
+    {
+        public int maxSubstances;
+        public int currentSubstancesCount;
```
Comment (Puchkof) [inline ChemistryLaboratory/ChemistryLaboratory/Class1.cs]: Make this property as private

**Review by Puchkof — APPROVED** [review]

**Pull request merged**